include/vcpkg/base/expected.h

```diff
@@ -213,6 +234,17 @@ namespace vcpkg
             }
         }
 
+        template<class F>
+        ExpectedT& map_error(F&& specific_error_generator)
```

Comment:
I don't know what the syntax you're using here is. I started with replace_error but changed it to this because I was concerned the word "replace" may signal that it replaces values.

Reply:
```cpp
ExpectedT<T, E2> ExpectedT<T, E1>::map_error(E2 (*f)(E1));
```

Reply:
>Isn't "replace" exactly what this does? It assigns the error if this is in an error state.

Yes, but the name "replace" also suggests that it could replace a value state.

Either way, I'm going to apply the suggested name.